Translation for the page 'Basics -> Forms'

deblasis opened the pull request:
Closes #23 

<!--

Thank you for the PR! Contributors like you keep React awesome!

Please see the Contribution Guide for guidelines:

https://github.com/reactjs/reactjs.org/blob/master/CONTRIBUTING.md

If your PR references an existing issue, please add the issue number below

-->

LucaBlackDragon commented on content/docs/forms.md:
````diff
@@ -9,30 +9,30 @@ redirect_from:
 ```
 
-This form has the default HTML form behavior of browsing to a new page when the user submits the form. If you want this behavior in React, it just works. But in most cases, it's convenient to have a JavaScript function that handles the submission of the form and has access to the data that the user entered into the form. The standard way to achieve this is with a technique called "controlled comp ...
+La sottomissione di questo form porta l'utente in una nuova pagina, questo è il comportamento di _default_ (di fabbrica). In React, se vuoi avere lo stesso comportamento, funziona. Ad ogni modo, potrebbe essere più conveniente avere una funzione JavaScript che gestisce la sottomissione del form e che ha accesso ai dati inseriti dall'utente. La tecnica standard con cui si può ottenere ciò prende i ...
````
```suggestion
Questo form si comporta come di consueto, facendo navigare l'utente in una nuova pagina quando viene inviato. In React, se vuoi avere lo stesso comportamento, non c'è bisogno di fare alcuna modifica. Ad ogni modo, potrebbe essere più conveniente avere una funzione JavaScript che gestisce l'invio del form e che ha accesso ai dati inseriti dall'utente. La tecnica standard con cui si può ottenere ciò prende il nome di "componenti controllati".
```

I hope it is a bit clearer this way

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -9,30 +9,30 @@ redirect_from:
+## Componenti Controllati {#controlled-components}
 
-In HTML, form elements such as `<input>`, `<textarea>`, and `<select>` typically maintain their own state and update it based on user input. In React, mutable state is typically kept in the state property of components, and only updated with [`setState()`](/docs/react-component.html#setstate).
+In HTML, gli eleenti di un form come `<input>`, `<textarea>` e `<select>` mantengono tipicamente il proprio stato e lo aggiornano in base all'input dell'utente. In React, lo stato mutabile viene tipicamente mantenuto nella proprietà `state` dei componenti eviene poi aggiornato solo mediante [`setState()`](/docs/react-component.html#setstate).
```
```suggestion
In HTML, gli elementi di un form come `<input>`, `<textarea>` e `<select>` mantengono tipicamente il proprio stato e lo aggiornano in base all'input dell'utente. In React, lo stato mutabile viene tipicamente mantenuto nella proprietà `state` dei componenti e viene poi aggiornato solo mediante [`setState()`](/docs/react-component.html#setstate).
```

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -9,30 +9,30 @@ redirect_from:
+In HTML, gli eleenti di un form come `<input>`, `<textarea>` e `<select>` mantengono tipicamente il proprio stato e lo aggiornano in base all'input dell'utente. In React, lo stato mutabile viene tipicamente mantenuto nella proprietà `state` dei componenti eviene poi aggiornato solo mediante [`setState()`](/docs/react-component.html#setstate).
 
-We can combine the two by making the React state be the "single source of truth". Then the React component that renders a form also controls what happens in that form on subsequent user input. An input form element whose value is controlled by React in this way is called a "controlled component".
+Possiamo combinare le due cose rendendo lo _state_ in React la "singola fonte attendibile" ([SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth)). Possiamo poi fare in modo che il componente React che renderizza il form controlli anche cosa succede all'interno del form in risposta agli input dell'utente. Un elemento di input in un form, il quale valore è controllato da React viene così ch ...
```
```suggestion
Possiamo combinare le due cose rendendo lo _state_ in React la "singola fonte attendibile" ([SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth)). Possiamo poi fare in modo che il componente React che renderizza il form controlli anche cosa succede all'interno del form in risposta agli input dell'utente. In un form, un elemento di input il cui valore è controllato da React viene così chiamato "componente controllato".
```

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -9,30 +9,30 @@ redirect_from:
+Possiamo combinare le due cose rendendo lo _state_ in React la "singola fonte attendibile" ([SSOT](https://en.wikipedia.org/wiki/Single_source_of_truth)). Possiamo poi fare in modo che il componente React che renderizza il form controlli anche cosa succede all'interno del form in risposta agli input dell'utente. Un elemento di input in un form, il quale valore è controllato da React viene così ch ...
 
-For example, if we want to make the previous example log the name when it is submitted, we can write the form as a controlled component:
+Ad esempio, se vogliiamo far si che l'esempio precedente registri il nome inserito, possiamo riscrivere il form sotto forma di componente controllato:
```
```suggestion
Ad esempio, se vogliamo far sì che l'esempio precedente registri il nome inserito, possiamo riscrivere il form sotto forma di componente controllato:
```

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -64,36 +64,38 @@ class NameForm extends React.Component {
+**[Prova su CodeSandbox](codesandbox://forms/1.js)**
 
-Since the `value` attribute is set on our form element, the displayed value will always be `this.state.value`, making the React state the source of truth. Since `handleChange` runs on every keystroke to update the React state, the displayed value will update as the user types.
+Dato che l'attributo `value` viene impostato nel nostro elemento form, il valore visualizzato sarà sempre `this.state.value`, rendendo lo stato in React l'unica fonte dati attendibile. Dato che la funzione `handleChange` viene eseguita ad ogni battitura per aggiornare lo stato di React, il valore visualizzato verrà aggiornato man mano che l'utente preme i tasti.
```
```suggestion
Dato che l'attributo `value` viene impostato nel nostro elemento form, il valore visualizzato sarà sempre `this.state.value`, rendendo lo stato in React l'unica fonte di dati attendibile. Dato che la funzione `handleChange` viene eseguita ad ogni battitura per aggiornare lo stato di React, il valore visualizzato verrà aggiornato man mano che l'utente preme i tasti.
```

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -105,15 +107,15 @@ class EssayForm extends React.Component {
 
   handleSubmit(event) {
-    alert('An essay was submitted: ' + this.state.value);
+    alert('Un tema è stato sottomesso: ' + this.state.value);
```
```suggestion
    alert('Un tema è stato inviato: ' + this.state.value);
```

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -178,121 +182,128 @@ class FlavorForm extends React.Component {
+**[Prova su CodeSandbox](codesandbox://forms/2.js)**
 
-Overall, this makes it so that `<input type="text">`, `<textarea>`, and `<select>` all work very similarly - they all accept a `value` attribute that you can use to implement a controlled component.
+Ricapitolando, ciò fa si che `<input type="text">`, `<textarea>` e `<select>` funzionino in modo molto simile - tutti accettano un attributo `value` che puoi utilizzare per implementare un componente controllato.
```
```suggestion
Ricapitolando, ciò fa sì che `<input type="text">`, `<textarea>` e `<select>` funzionino in modo molto simile - tutti accettano un attributo `value` che puoi utilizzare per implementare un componente controllato.
```

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -64,36 +64,38 @@ class NameForm extends React.Component {
-  Hello there, this is some text in a text area
+  Nel mezzo del cammin di nostra vita
+  mi ritrovai per una selva oscura
+  ché la diritta via era smarrita.
```
💯

deblasis replied:
🤓 😄

LucaBlackDragon commented on content/docs/forms.md:
````diff
@@ -178,121 +182,128 @@ class FlavorForm extends React.Component {
 ```
 
-Because its value is read-only, it is an **uncontrolled** component in React. It is discussed together with other uncontrolled components [later in the documentation](/docs/uncontrolled-components.html#the-file-input-tag).
+Dato che il suo valore è sola-lettura, è un componente **non controllato** in React. Riprenderemo il discorso riguardo questo ed altri componenti non controllati [in seguito](/docs/uncontrolled-components.html#the-file-input-tag).
````
```suggestion
Dato che il suo valore è in sola-lettura, è un componente **non controllato** in React. Riprenderemo il discorso riguardo questo ed altri componenti non controllati [in seguito](/docs/uncontrolled-components.html#the-file-input-tag).
```

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -178,121 +182,128 @@ class FlavorForm extends React.Component {
+## Gestione di Input Multipli {#handling-multiple-inputs}
 
-When you need to handle multiple controlled `input` elements, you can add a `name` attribute to each element and let the handler function choose what to do based on the value of `event.target.name`.
+Quando devi gestire diversi elementi `input`, puoi aggiungere un attributo `name` ad ognuno di essi e far si che la funzione handler controlli cosa fare in base al valore di `event.target.name`.
```
```suggestion
Quando devi gestire diversi elementi `input`, puoi aggiungere un attributo `name` ad ognuno di essi e far sì che la funzione handler controlli cosa fare in base al valore di `event.target.name`.
```

LucaBlackDragon commented on content/docs/forms.md:
```diff
@@ -178,121 +182,128 @@ class FlavorForm extends React.Component {
+## Valore Null in Input Controllati {#controlled-input-null-value}
 
-Specifying the value prop on a [controlled component](/docs/forms.html#controlled-components) prevents the user from changing the input unless you desire so. If you've specified a `value` but the input is still editable, you may have accidentally set `value` to `undefined` or `null`.
+Specificando la prop `value` in un [componente controllato](/docs/forms.html#controlled-components) può far si che l'utente non possa cambiare l'input, a meno che tu lo desideri. Se hai specificato un `value` ma l'input è ancora editabile, potresti aver accidentalmente impostato `value` come `undefined` o `null`.
```
```suggestion
Specificare la prop `value` in un [componente controllato](/docs/forms.html#controlled-components) fa sì che l'utente possa cambiare l'input solo quando lo desideri. Se hai specificato un `value` ma l'input è ancora editabile, potresti aver accidentalmente impostato `value` come `undefined` o `null`.
```

deblasis commented:
> I have fixed some typos and replaced all the occurences of "sottomettere" with "inviare", because "Invia" is by far the most common translation of "Submit" in form buttons.
> This is true not only in a technical context; for example, when I tell my colleagues that last month I submitted a transfer of residence form, I say "il mese scorso ho inviato il modulo per il cambio di residenza" (by the way, is it actually called "transfer of residence form" in the UK? I'm just guessing!).

You are right, `Invia` is way better!

yep, it's called "transfer of residence form". BTW, slightly off-topic, with the Brexit thing going on, I believe that fewer and fewer people are using it to come over here 🤔 at least until things are clearer

LucaBlackDragon commented:
> 
> 
> > I have fixed some typos and replaced all the occurences of "sottomettere" with "inviare", because "Invia" is by far the most common translation of "Submit" in form buttons.
> > This is true not only in a technical context; for example, when I tell my colleagues that last month I submitted a transfer of residence form, I say "il mese scorso ho inviato il modulo per il cambio di residenza" (by the way, is it actually called "transfer of residence form" in the UK? I'm just guessing!).
> 
> You are right, `Invia` is way better!
> 
> yep, it's called "transfer of residence form". BTW, slightly off-topic, with the Brexit thing going on, I believe that fewer and fewer people are using it to come over here 🤔 at least until things are clearer

I think we'll have to see what happens in the next months.
I've never been in the UK for other reasons than holidays or improving my language, but I LOVE that country and this whole Brexit thing is very sad 😞
